scanner/templates/node/Dockerfile

```diff
@@ -29,25 +24,23 @@ RUN {{ .packager }} install && {{ .packager }} run build
 RUN {{ .packager }} install
 {{ end }}
 
-FROM debian:bullseye-slim
+FROM node:${NODE_VERSION}-slim
```

Comment:
What about building upon previous steps, like we do in the current `remix`/`remix_prisma` templates?

That way we don't need to re-set the actual base image over and over again

Reply:
I agree with more steps.  mkdir and WORKDIR should be moved to base.

I don't (yet?) see the value of separate deps, production-deps, and build.  Can you explain why they are separate steps?  A see a lot of copying and repetition.  Making a single streamlined build step seems to me to be more readable?

Unrelated, but perhaps `fly launch` should generate a 64 character hex string and run `fly secrets set SESSION_SECRET`?

Reply:
> Can you explain why they are separate steps?

I'm not the author of the original `Dockerfile`, so I don't know the thought process behind it.
According to https://github.com/superfly/flyctl/pull/569 & [Remix-related PRs](https://github.com/superfly/flyctl/pulls?q=is%3Apr+Remix), @jsierles is the original author of the file & main maintainer of the Remix templates.

> I see a lot of copying and repetition

I think this is because
- the production image only needs a clean image + the build files & production deps
- the build step needs a clean image + all dependencies
- the production deps needs  a clean image + all dependencies to remove dev dependencies

> Making a single streamlined build step seems to me to be more readable?

I'm in favor of simplifying this, but I don't think we can as we don't want to end up with unnecessary `node_modules` in the production image 🤔

> Unrelated, but perhaps `fly launch` should generate a 64 character hex string and run `fly secrets set SESSION_SECRET`?

Should `fly launch` also set this value in `.env`?
Or log it so it can be re-used there?

I guess this is something that should be done regardless the scanner that's used?

Reply:
> Do we even need the LABEL?

No, but it helps us here at fly.io prioritize where we focus our efforts.  It's not perfect, but going forward it will enable us to scan for applications that use Remix.